rpc/src/v1/helpers/nonce.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::sync::Arc;
+use futures::{Future, Poll, Async};
+use util::{U256, RwLock};
+use std::collections::{VecDeque};
+
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub enum NonceError {
+    Dropped,
+    InvalidPoll,
+}
+
+#[derive(Copy, Clone)]
+enum NonceState {
+    Prospective,
+    Reserved,
+    Dispatch,
+    Error(NonceError),
+}
+
+pub struct Nonce {
+    state: NonceState,
+    value: U256,
+    reserved: Arc<RwLock<VecDeque<U256>>>,
```

Comment:
Issue there is how to check that the nonce is the oldest (ie. front of the queue) before dispatching.  Otherwise if it is dispatched and then an older reserved nonce is dropped it would create a gap, right?